File: 49_Group_Anagrams.py
from typing import List 
class Solution: 
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]: 
        anagram = {} 
        for word in strs:
            sorted_word = "".join(sorted(word))
            if sorted_word in anagram:
                anagram.get(sorted_word).append(word)
            else:
                anagram[sorted_word] = [word] 
        return [word for word in anagram.values()]

File: cpp/49_Group_Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> sorted;
        vector<vector<string>> result;
        for(auto &str : strs){
            string key = str;
            sort(key.begin(),key.end());
            sorted[key].push_back(str);
        }

        for(auto &entry : sorted){
            result.push_back(entry.second);
        }
        return result;
    }
};